pkg/memo/memo.go

```diff
@@ -0,0 +1,73 @@
+package memo
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+// InboundMemo represents the inbound memo structure for non-EVM chains
```

Comment:
> InboundMemo represents the inbound **memo structure for non-EVM chains**

I think we should rename this a bit because  realistically this format will be used only for BTC / UTXO chains where we don't have smart contracts at all

Reply:
>Solana and TON will use some similar structs (per @lumtis).

As discussed on Slack, not TON nor Solana will use it but I personally do see the issue with InboundMemo terminology, I think the term is good

Reply:
Yeah. SOL and TON will use their own structs defined in gateway.